### PR TITLE
Skipchain

### PR DESCRIPTION
**Pull request by ineiti**
First implementation of skipchains

### CODE REVIEW

**Comment by Liamsi** [comment]
I put it into Ready4Merge pipeline. We will use this code for every application. Please review (/cc @ineiti @nikkolasg and myself)

File: protocols/skipchain/skipchain.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		return nil, errors.New("couldn't register propagete-handler: " + err.Error())
+	}
+	return Skipchain, nil
+}
+
+// Start the protocol
+func (p *ProtocolSkipchain) Start() error {
+	dbg.Lvl3("Starting Skipchain")
+	block := &SkipBlock{Index: 0, X0: p.TreeNode().PublicAggregateSubTree, Nodes: p.Tree().List()}
+	//p.LastBlock = block.Hash()
```
Comment (ineiti) [inline protocols/skipchain/skipchain.go]: If it is not used, remove.

File: protocols/skipchain/skipchain_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	"time"
+)
+
+// Tests a 5-node system
+func TestNode(t *testing.T) {
+	dbg.TestOutput(testing.Verbose(), 4)
+	local := sda.NewLocalTest()
+	nbrNodes := 2
+	_, _, tree := local.GenTree(nbrNodes, false, true, true)
+	//dbg.Lvl3(tree.Dump())
```
Comment (ineiti) [inline protocols/skipchain/skipchain_test.go]: Put it to Lvl4 or so, but don't comment it.

**Comment by ineiti** [comment]
I'm not sure it is ready to be merged. I'd like to start making a service out of it this Friday and see what is still missing. What do the others think?

**Comment by ineiti** [comment]
Superseded by #357

**Pull request closed without merging**